files/default/neutron-ha-tool.py

```
@@ -654,10 +654,14 @@ def migrate_router(qclient, router_id, agent, target,
     qclient.remove_router_from_l3_agent(agent['id'], router_id)
     LOG.debug("Removed router from agent=%s" % agent['id'])
 
-    # ensure it is removed or log an error
+    # ensure it is removed
     if router_id in list_routers_on_l3_agent(qclient, agent['id']):
-        raise RuntimeError("Failed to remove router_id=%s from agent_id=%s" %
-                           (router_id, agent['id']))
+        qclient.remove_router_from_l3_agent(agent['id'], router_id)
+        LOG.debug("Removed (again) router from agent=%s" % agent['id'])
```

Not sure that I like the workaround : ). Actually the resource agent is retrying that, and when I did some manual test this worked. So that means that if I am not wrong, the first delete raised and exception somewhere.

But if it is not raising anything, I have two comments:

* Add a reference of the bsc# bug report in the comment in the code and in the commit. Be clear in the code that this is a workaround that needs to be reverted once the real bug is fixed (or SOC6 dies)

* I wonder if "Removed (again)" is a bit misleading here. Maybe a WARNING, instead of a DEBUG and with a different text, like "The router was not correctly deleted, retrying" or something

---

> Actually the resource agent is retrying that, and when I did some manual test this worked. So that means that if I am not wrong, the first delete raised and exception somewhere.

During the upgrade we're not calling neutron-ha-tool from the resource agent. And the behaviour of ```--retry``` actually interferes with other requirements of the upgrade as it would also retry in cases where we need to raise an error during the ugprade (see https://bugzilla.suse.com/show_bug.cgi?id=1020914 that's why we also introduced: https://github.com/SUSE-Cloud/cookbook-openstack-network/pull/20 ).

---

@aplanas I don't like the workaround neither but the fix seems far down the road, because heavy backport is needed.

The retry mechanism you mention does not solve the problem here. The first (and incomplete) delete does not produce an error thus the retry mechanism is not activated.

About your two comments:
- I'll add a reference to the bsc#, good point.
- I can change the output on the retry. I like the new text you proposed. I don't think though it's a warning since it's expected behavior for cloud6.